### PR TITLE
pythonPackages.crate: mark broken

### PR DESCRIPTION
**Pull request by layus**
###### Motivation for this change

csvkit was broken because "crate" python package does not (yet) support SQLAlchemy >= 1.4 that nixpkgs ships since 9974c90a1597a7b2b009e798374615899d57db7d.

Mark "crate" broken for now, and disable tests in agate-sql, as "crate" is only used for tests.

Use a complex, opinionated mechanism to force a re-evaluation of this strategy when "crate" gets updated.

and voilà: csvkit works again.

From a quick analysis, this crate package is only used in agate-sql. Hence the long time to fix.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

I was really light on testing the impact on other packages, but given that the build is broken, marking the package as broken should not have such an incredible impact.

### CODE REVIEW

**Comment by layus** [comment]
/cc @vrthra @doronbehar for info.

@ryantm What do you think of the convoluted assert mechanism. Does it make any sense, or does it get in the way of nixpkgs-update ?

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 130017` at 7563744e run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built successfully:</summary>
  <ul>
    <li>csvkit</li>
    <li>python38Packages.agate-sql</li>
    <li>python39Packages.agate-sql</li>
  </ul>
</details>
<details>
  <summary>3 suggestions:</summary>
  <ul>
    <li>warning: unused-argument

Unused argument: `nose`.
Near pkgs/development/python-modules/agate-sql/default.nix:8:3:
```
  |
8 | , nose
  |   ^
```
</li>
    <li>warning: unused-argument

Unused argument: `crate`.
Near pkgs/development/python-modules/agate-sql/default.nix:7:3:
```
  |
7 | , crate
  |   ^
```
</li>
    <li>warning: unused-argument

Unused argument: `geojson`.
Near pkgs/development/python-modules/agate-sql/default.nix:9:3:
```
  |
9 | , geojson
  |   ^
```
</li>
  </ul>
</details>

File: pkgs/development/python-modules/agate-sql/default.nix [hunk 1]
```diff
@@ -22,7 +22,10 @@ buildPythonPackage rec {
 
   propagatedBuildInputs = [ agate sqlalchemy ];
 
-  checkInputs = [ crate nose geojson ];
+  # crate is broken with SQLAlchemy > 1.3
+  # These tests rely on crate, and make the whole build fail
+  #checkInputs = [ crate nose geojson ];
+  doCheck = false;
```
Comment (jonringer) [inline pkgs/development/python-modules/agate-sql/default.nix:28]: checkInputs are effectful if checkPhase is disabled
```suggestion
  # crate is broken with SQLAlchemy > 1.3
  # These tests rely on crate, and make the whole build fail
  doCheck = false;
  checkInputs = [ crate nose geojson ];
 
```

Comment (layus) [reply pkgs/development/python-modules/agate-sql/default.nix]: Indeed, fixed. Thanks :)

**Comment by doronbehar** [comment]
I approve the changes here after squash.

**Comment by SuperSandro2000** [comment]
> I approve the changes here after squash.

Just squash merge it.

**Pull request merged**